### PR TITLE
Fix solariumphp/solarium#819 Getting named facet from a JsonQuery using getFacet('local_key_value') throws TypeError  

### PR DESCRIPTION
**Pull request by ee-dev**
Add FacetResultInterface to allow Solarium\Component\Result\FacetSet to access to a JsonQuery result by it's key in FacetSet::getFacet()
Includes JsonQuery example.

### CODE REVIEW

File: examples/index.html [hunk 1]
```diff
@@ -56,6 +56,7 @@ <h2>Examples</h2>
                             <li><a href="2.1.5.1.5-facet-pivot.php">2.1.5.1.5 Facet pivot</a></li>
                             <li><a href="2.1.5.1.6-facet-interval.php">2.1.5.1.6 Facet interval</a></li>
                             <li><a href="2.1.5.1.7-facet-json-terms.php">2.1.5.1.7 Facet json terms</a></li>
+                            <li><a href="2.1.5.1.8-facet-json-query">2.1.5.1.8 Facet json query</a></li>
```
Comment (thomascorthals) [inline examples/index.html:59]: Can you add `.php` to the link? It depends on the webserver config if it'll work without the extension.

```suggestion
                            <li><a href="2.1.5.1.8-facet-json-query.php">2.1.5.1.8 Facet json query</a></li>
```

File: examples/2.1.5.1.8-facet-json-query.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// this executes the query and returns the result
+$resultset = $client->select($query);
+
+// display the total number of documents found by solr
+echo 'NumFound: '.$resultset->getNumFound();
+
+// JsonQuery returns a FacetSet, not Buckets like a jsonTerms facet
+$stock_query = $resultset->getFacetSet()->getFacet('stock_query');
+
+// the facet count is returned as a 'count' sub-facet
```
Comment (mkalkbrenner) [inline examples/2.1.5.1.8-facet-json-query.php:30]: In search_api_solr we use this comment:
    
```
// The implicit "count" aggregation over all results matching the query
// exists only any JSONFacet set.
```

I think we should comment like this:
```
// A JSONFacet triggers an implicit "count" aggregation over all results which is accessible
// just like a Facet within the FacetSet.
```

Comment (ee-dev) [reply examples/2.1.5.1.8-facet-json-query.php:30]: Ah, except I'm not sure 'all results' is quite the right phrase as this is the count for the JsonQuery alone, not all results?
Perhaps:
// A JsonQuery triggers an implicit "count" aggregation for the result of the query which is accessible just like a Facet within the FacetSet.

Comment (mkalkbrenner) [reply examples/2.1.5.1.8-facet-json-query.php:30]: sounds good :-)

**Pull request merged**